### ISSUE TITLE
Small problem with max depth of nesting

### ISSUE THREAD

**Issue by mbatyumov** [brigade/scss-lint#748]
In sass from version 3.3.0 we can to add suffixes (for example: &-suffix or &_suffix).

And if I have the next constraint:

``` yml
NestingDepth:
  enabled: true
  max_depth: 3
  ignore_parent_selectors: false
```

I'll get an error with this code:

``` scss
.icon {
    &-user {
        &-small {
            &:hover {}
        }
    }
}
```

But after proccessing it looks great:

``` css
.icon-user-small:hover {}
```

if I set max_depth to 4, I can miss real errors. Is it possible to solve this?

**Comment by sds** [comment]
The linter is working as expected here. `NestingDepth` doesn't refer to selector depth (that's what the `SelectorDepth` linter is for), it refers to the literal syntactic nesting you can do with Sass. In your case, your `&:hover` _is_ at depth 4.

The rationale here is that the more nested your code is the harder it is to understand. Your example is a simple one, but it's quite easy to find examples where you are deep in a nested block and have to scroll-up to figure out what the parent reference `&` is actually referring to.

If you don't like the false positives, disable it for the file in question, or disable the linter entirely.

Hope that helps.

**Comment by mbatyumov** [comment]
Thank you, Shane! It's realy helpful. But if I disable the linter `NestingDepth`, who will highlight this?

``` scss
.selector {
    &.monster {
        &.with {
            &.a-lot-of {
                &.classes {}
            }
        }
    }
}
```

**Comment by sds** [comment]
Chained classes are reported by the [`ChainedClasses`](https://github.com/brigade/scss-lint/blob/master/lib/scss_lint/linter/chained_classes.rb) linter. Perhaps it could be enhanced to support a configurable maximum number of chained classes before reporting an error.

If that's something you'd like, feel free to open another issue/pull request!

**Comment by sds** [comment]
Ah, I just looked at the implementation and realized it doesn't actually handle the case you describe.

I'm not sure what to offer here. Disabling `NestingDepth` necessarily has its limitations. It's very difficult to resolve `&` selectors in the general case, especially when dealing with comma sequences. I'm definitely open to a well-tested pull request that solves this, however!
